dpnp/backend/extensions/ufunc/elementwise_functions/interpolate.cpp

```diff
@@ -0,0 +1,296 @@
+                                            const void *,      // right
+                                            void *,            // out
+                                            const std::size_t, // n
+                                            const std::size_t, // xp_size
+                                            const std::vector<sycl::event> &);
+
+template <typename T, typename TIdx = std::int64_t>
+sycl::event interpolate_call(sycl::queue &exec_q,
+                             const void *vx,
+                             const void *vidx,
+                             const void *vxp,
+                             const void *vfp,
+                             const void *vleft,
+                             const void *vright,
+                             void *vout,
+                             const std::size_t n,
+                             const std::size_t xp_size,
+                             const std::vector<sycl::event> &depends)
+{
+    using type_utils::is_complex_v;
+    using TCoord = std::conditional_t<is_complex_v<T>, value_type_of_t<T>, T>;
+
+    const TCoord *x = static_cast<const TCoord *>(vx);
+    const TIdx *idx = static_cast<const TIdx *>(vidx);
+    const TCoord *xp = static_cast<const TCoord *>(vxp);
+    const T *fp = static_cast<const T *>(vfp);
+    const T *left = static_cast<const T *>(vleft);
+    const T *right = static_cast<const T *>(vright);
+    T *out = static_cast<T *>(vout);
+
+    using dpnp::kernels::interpolate::interpolate_impl;
+    sycl::event interpolate_ev = interpolate_impl<TCoord, T>(
+        exec_q, x, idx, xp, fp, left, right, out, n, xp_size, depends);
+
+    return interpolate_ev;
+}
+
+interpolate_fn_ptr_t interpolate_dispatch_vector[td_ns::num_types];
+
+void common_interpolate_checks(
+    const dpctl::tensor::usm_ndarray &x,
+    const dpctl::tensor::usm_ndarray &idx,
+    const dpctl::tensor::usm_ndarray &xp,
+    const dpctl::tensor::usm_ndarray &fp,
+    const dpctl::tensor::usm_ndarray &out,
+    const std::optional<const dpctl::tensor::usm_ndarray> &left,
+    const std::optional<const dpctl::tensor::usm_ndarray> &right)
+{
+    array_names names = {{&x, "x"}, {&xp, "xp"}, {&fp, "fp"}, {&out, "out"}};
+
+    auto array_types = td_ns::usm_ndarray_types();
+    int x_type_id = array_types.typenum_to_lookup_id(x.get_typenum());
+    int idx_type_id = array_types.typenum_to_lookup_id(idx.get_typenum());
+    int xp_type_id = array_types.typenum_to_lookup_id(xp.get_typenum());
+    int fp_type_id = array_types.typenum_to_lookup_id(fp.get_typenum());
+    int out_type_id = array_types.typenum_to_lookup_id(out.get_typenum());
+
+    if (x_type_id != xp_type_id) {
+        throw py::value_error("x and xp must have the same dtype");
+    }
+    if (fp_type_id != out_type_id) {
+        throw py::value_error("fp and out must have the same dtype");
+    }
+    if (idx_type_id != static_cast<int>(td_ns::typenum_t::INT64)) {
```

Comment:
You probably could add function to `validation utils`. Something like
```C++
void check_has_dtype(const array_ptr &arr,
    const typenum_t dtype,
    const array_names &names);
```

And call it here

Reply:
done, thank you